### ISSUE TITLE
Remove participant button is before the list box in tab order

### ISSUE THREAD

**Issue by daniel-montalvo** [w3c/w3c-website#269]
"Remove participant" button is before the list box in tab order

* Add new calendar event
* Select participants tab
* Add a participant using one of the three lists
* Then remove participant

The remove button is after the lisst box in tab order

At the moment, to find the remove button, I have to shift tab from the edit box

**Comment by mcking65** [comment]
I wonder how the add field is visually positioned relative to the list of organizers or participants. As a screen reader user, it appears that the add input is to the right of or below the list.

It seems to me that it would be better for all users if the add input were positioned to the left or above the list.

I also noticed that focus is lost when removing a participant. It should be placed on the next participant in the list, or if none remain, on the add input.

**Comment by deniak** [comment]
@daniel-montalvo @mcking65, I started a mockup to test a fix to this issue. Can you please go to https://mir.w3.org/~denis/add-event.html and see if this solves the problem?

**Comment by daniel-montalvo** [comment]
Hi @deniak 

The focus loss when removing a partiicpant seems to have been already resolved

However, when you focus the edit field, type some characters, do down arrow key and press enter to select a participant, you still have to press shift tab to get to the "Remove participant" button.

**Comment by deniak** [comment]
> However, when you focus the edit field, type some characters, do down arrow key and press enter to select a participant, you still have to press shift tab to get to the "Remove participant" button.

I believe that's the expected behavior. This way you can add new participants directly. If the focus is on the button, you will have to press tab to go to the input.

**Comment by daniel-montalvo** [comment]
@deniak 

Thanks for your quick response.

1. I would agree that focus should be put in the edit field after adding a participant. The alternative would be highly inefficient.
2. The issue appears when you want to remove a participant. Having to press Shift tab from the edit field to the remove button is weird. These buttons are new content that appears on screen without the user necessarily knowing about it. We should expect that the user will continue to navigate forward through the page, either by using tab or by using their screen reader "move forward"-like functionality.

**Comment by deniak** [comment]
@daniel-montalvo can you please check again the mockup? The list of selected users should now appear *after* the input.

**Comment by mcking65** [comment]
@deniak 

The mockup is fabulous!! do that! After adding someone, I can tab forward to the remove buttons. Activating remove also properly places focus. Very nice.

**Comment by deniak** [comment]
The fix has been deployed by @jean-gui.

**Comment by Limpisey77922321** [comment]
verified

**Comment by Limpisey77922321** [comment]
> "Remove participant" button is before the list box in tab order
> 
> * Add new calendar event
> * Select participants tab
> * Add a participant using one of the three lists
> * Then remove participant
> 
> The remove button is after the lisst box in tab order
> 
> At the moment, to find the remove button, I have to shift tab from the edit box
